### PR TITLE
Add RBAC authorization

### PR DESCRIPTION
**Pull request by yashmurty**
- Basic RBAC implementation (https://docs.nestjs.com/security/authorization#basic-rbac-implementation)
- Minor refactor on JWT claims type.

### CODE REVIEW

File: src/auth/dtos/auth-register-output.dto.ts [hunk 1]
```diff
@@ -7,4 +7,6 @@ export class RegisterOutput {
   name: string;
   @ApiProperty()
   username: string;
+  @ApiProperty()
+  roles: string[];
```
Comment (asad-mlbd) [inline src/auth/dtos/auth-register-output.dto.ts:11]: roles: ROLE[]

File: src/auth/dtos/auth-token-output.dto.ts [hunk 2]
```diff
@@ -8,6 +8,12 @@ export class AuthTokenOutput {
   refreshToken: string;
 }
 
-export class TokenUserIdentity {
+export class UserAccessTokenClaims {
+  id: number;
+  username: string;
+  roles: string[];
+}
```
Comment (asad-mlbd) [inline src/auth/dtos/auth-token-output.dto.ts:15]: roles: ROLE[]

**Review by asad-mlbd — APPROVED** [review]
In most cases, we used `roles: string[]`, maybe better to use roles: ROLE[]

**Review by sivanpayyadakath — APPROVED** [review]

File: src/user/dtos/user-create-input.dto.ts [hunk 3]
```diff
@@ -11,4 +11,8 @@ export class CreateUserInput {
   @IsNotEmpty()
   @IsString()
   password: string;
+
+  @IsNotEmpty()
+  @IsString()
```
Comment (sivanpayyadakath) [inline src/user/dtos/user-create-input.dto.ts:16]: @yashmurty 
`@IsString({each: true})`
this is the correct one I guess ?  🤔

File: src/auth/auth.module.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { Module } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import { PassportModule } from '@nestjs/passport';
 import { JwtModule } from '@nestjs/jwt';
+import { APP_GUARD } from '@nestjs/core';
```
Comment (sivanpayyadakath) [inline src/auth/auth.module.ts:5]: unused import !!!!!!!!!

File: src/user/dtos/user-create-input.dto.ts (hunk 3 above)
Comment (yashmurty) [reply src/user/dtos/user-create-input.dto.ts:18]: Changing it to `@IsArray`.
Thanks for pointing it out.

**Pull request merged**